fetchurl: fix and add extra KDE mirrors

bhipple opened the pull request:
- The gwdg.de mirror has moved the relative path of its KDE tarballs
- Add new mirrors from Berkeley and MIT, which are on the list of officially supported mirrors

https://download.kde.org/extra/download-mirrors.html

###### Motivation for this change
I had some issues downloading KDE source tarballs with the existing mirrors; at least 1 of them is broken, and fixed here.  The newly added ones work.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

bhipple committed:
fetchurl: fix and add extra KDE mirrors

- The gwdg.de mirror has moved the relative path of its KDE tarballs
- Add new mirrors from Berkeley and MIT, which are on the list of officially supported mirrors

https://download.kde.org/extra/download-mirrors.html

c0bw3b commented on pkgs/build-support/fetchurl/mirrors.nix:
```diff
@@ -73,7 +73,9 @@ rec {
   # Mirrors of ftp://ftp.kde.org/pub/kde/.
   kde = [
     "http://download.kde.org/download.php?url="
```
Could we also do
```suggestion
    "https://download.kde.org/download.php?url="
```

Note that it's a redirect anyway:
```shell
> http -h http://download.kde.org/download.php
HTTP/1.1 301 Moved Permanently
[...]
Location: https://download.kde.org/download.php
```

c0bw3b commented:
Also consider adding one HTTP mirror from Asia for users on this side of the world?

bhipple committed:
More KDE mirror touchups

- The official one is a redirect to HTTPS anyways, so go directly to HTTPS
- Add China USTC for Asian users
- Swap Finland university from FTP to HTTP

bhipple commented on pkgs/build-support/fetchurl/mirrors.nix:
```diff
@@ -70,13 +70,14 @@ rec {
-    "http://download.kde.org/download.php?url="
-    http://ftp.gwdg.de/pub/x11/kde/
-    ftp://ftp.heanet.ie/mirrors/ftp.kde.org/
-    ftp://ftp.kde.org/pub/kde/
```
This one's redundant, since we already have the HTTPS kde.org

bhipple commented on pkgs/build-support/fetchurl/mirrors.nix:
```diff
@@ -70,13 +70,14 @@ rec {
   kde = [
-    "http://download.kde.org/download.php?url="
-    http://ftp.gwdg.de/pub/x11/kde/
-    ftp://ftp.heanet.ie/mirrors/ftp.kde.org/
```
This one does not appear to be online, as far as I can tell; removed

bhipple commented on pkgs/build-support/fetchurl/mirrors.nix:
```diff
@@ -70,13 +70,14 @@ rec {
-    http://ftp.gwdg.de/pub/x11/kde/
-    ftp://ftp.heanet.ie/mirrors/ftp.kde.org/
-    ftp://ftp.kde.org/pub/kde/
-    ftp://ftp.funet.fi/pub/mirrors/ftp.kde.org/pub/kde/
```
Swapped to HTTP, which is likely faster, and more likely to make it through firewalls.

c0bw3b commented:
LGTM.
I would maybe have kept one FTP mirror at the end of the list, for the unlikely scenario where everyone's httpd goes unreachable but ftp daemons are still up. But minor nitpick.

c0bw3b committed:
fetchurl: keep one ftp KDE mirror
